### PR TITLE
Add password generation for hashi_vault lookup

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY

Extend hashi_vault lookup to allow generate secrets if not exists. This functionality is similar to password lookup and allow us to change all local passwords to vault backend. 

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

lookup: hashi_vault

### CODE REVIEW

**Comment by juliedavila** [comment]
I'm not really in favor of this. A `lookup` plugin imo should do just that, look things up. When we talk about changing the state of the system, in the case a vault secret, then I think a proper module would be the route that should be taken.

If you think about the `password` lookup, even that doesn't change any state, it's merely used to generate a credential which you can store and subsequently use elsewhere.

**Review by juliedavila — COMMENTED** [review]
To add further examples if we look at the upstream docs https://docs.ansible.com/ansible/latest/plugins/lookup.html I dont think any lookup plugin changes the state on the data source it queries

File: lib/ansible/plugins/lookup/hashi_vault.py [hunk 1]
```diff
@@ -9,11 +9,11 @@
   lookup: hashi_vault
   author: Jonathan Davila <jdavila(at)ansible.com>
   version_added: "2.0"
-  short_description: retrieve secrets from HashiCorp's vault
+  short_description: retrieve secrets from HasihCorp's vault
```
Comment (achinthagunasekara) [inline lib/ansible/plugins/lookup/hashi_vault.py]: I think it's spelled `HashiCorp`?

**Comment by ghost** [comment]
We just change from store passwords in our filesystem to Vault and we need to replace `lookup` without change logic.

File: lib/ansible/plugins/lookup/hashi_vault.py (hunk 1 above)
Comment (ghost) [reply lib/ansible/plugins/lookup/hashi_vault.py]: Corrected

**Comment by juliedavila** [comment]
@AgilDat I'm not sure what you mean by `we need to replace lookup without change logic.`. Lookup plugins shouldn't be making writes to anything, which this does.

**Comment by ghost** [comment]
@defionscode if yours lookups don't write is ok for me, just close this pull request, I just share my work.
It's simpler for me just to make Vault lookup to work as password lookup, just that. For  my project this has a lot of sense.

**Comment by juliedavila** [comment]
Ok. Like I said earlier, I would welcome a PR for a module that does write/update operations, it's just not something that should be done with a lookup plugin.

**Pull request closed without merging**